src/components/Personalization/in-app-message-actions/actions/displayModal.js

```diff
@@ -0,0 +1,102 @@
+import { addStyle, removeElements } from "../utils";
+
+const STYLE_TAG_ID = "alloy-messaging-modal-styles";
+const MODAL_CSS_CLASSNAME = "alloy-modal";
+
+const closeModal = () => {
+  removeElements(MODAL_CSS_CLASSNAME);
+};
+const showModal = ({ buttons = [], content }) => {
+  removeElements(MODAL_CSS_CLASSNAME);
+
+  addStyle(
+    STYLE_TAG_ID,
+    `.alloy-modal {
+      display: none;
+      width: 100%;
+      height: 100%;
+      position: fixed;
+      left: 0;
+      top: 0;
+      background: rgba(0, 0, 0, 0.3);
+      z-index: 999;
+    }
+    .alloy-modal--show { display: flex; }
+    .alloy-align-center {
+      justify-content: center;
+    }
+    .alloy-align-vertical {
+      align-items: center;
+    }
+    .alloy-modal-container {
+      position: relative;
+      width: 100%;
+      max-width: 600px;
+      max-height: 800px;
+      padding: 20px;
+      margin: 12px;
+      background: #fff;
+    }
+    .alloy-modal-content {
+      margin: 15px 0;
+      vertical-align: top;
+      text-align: left;
+    }
+    .alloy-modal-close--x {
+      font-size: 30px;
+      position: absolute;
+      top: 3px;
+      right: 10px;
+    }
+    .alloy-modal-close--x:hover {
+      cursor: pointer;
+    }
+    .alloy-modal-buttons button {
+      margin-right: 5px;
+    }`
+  );
+
+  const modal = document.createElement("div");
+  modal.className = `${MODAL_CSS_CLASSNAME} alloy-align-center alloy-align-vertical alloy-modal--show`;
+
+  const modalContainer = document.createElement("div");
+  modalContainer.className = "alloy-modal-container";
+
+  const closeButton = document.createElement("a");
```

Comment:
I'd prefer this to be a button, since it doesn't result in a URL change. It's a semantic preference.

Here's some css to get you started

```css
.alloy-modal-close--x {
  margin: 0;
  padding: 0;
  border: none;
  font-family: inherit;
  background: none;
  cursor: pointer;
  outline: none;
  
  font-size: 30px;
  position: absolute;
  top: 3px;
  right: 10px;
}
```

I'd also like to see all the classes as constants in string variables.

Reply:
Thanks for the feedback @carterworks!  I agree with your suggestion.  For now the modal and banner message rendering code has just been stubbed in to prove the concept.  We are waiting for product specs and then we'll polish it up to match the mocks and make it production grade. 😸